vassal-app/src/main/java/VASSAL/tools/ErrorDialog.java

```diff
@@ -242,6 +243,17 @@ public static void bug(final Throwable thrown) {
     );
   }
 
+  public static Future<?> showFileOverwriteFailure( // I don't see why it shouldn't have a meaningful name? So much less confusing.
+    FileSystemException thrown,
+    String details,
+    String messageKey,
+    Object... args) {
+
+    return ProblemDialog.showFileOverwriteFailure(
```

Comment:
Is there a reason for putting the implementation into ProblemDialog? 

ProblemDialog was intended to have general "show stuff" methods in it, and ErrorDialog slightly more specific "error flavored" dialogs. Can the implementation of all of this go into WriteErrorDialog and leave ErrorDialog and ProblemDialog unmodified?

Reply:
I'm pretty sure not, because we both need to pass the "two filenames" thing down, *and* be able to detect that we're in a situation where we need to do the two-filename-thing. I believe I simply followed the chain down to the earliest point that I could actually make the thing happen (or rather, I followed it too far, and then backed up).

Now I would be WHOLLY in favor of having WriteErrorDialog simply _skip_ ErrorDialog (which as far as I can tell contributes nothing to this case except an extra maze-like layer) and directly call ProblemDialog, if that seems to make any sense. 

This has been my first real "adventure" in this section of the code (the error dialog maze), and I do feel like the maze-like-nature of it discourages decent error reporting -- as evidenced for example in various decisions that clearly got made by somebody reporting earlier versions of these same errors I'm dealing with here. I think we would be a lot better off if the methods weren't e.g. all called "showDetails", which neither helps someone coming in and looking to see if any appropriate error reporting methods already exist nor helps someone who knows they need to add a new one. I'm sure it started off well-intentioned, but it doesn't seem to be serving "error reporting customers" very well. It kind of straight-jackets everything into a one-size-fits-all, which I guess works okay for the generic "whoops it crashed, give up now" situation, but anything custom has to be threaded through several different layers and that has clearly discouraged people from doing it.

Reply:
Prior to my trying to rationalize the various error dialogs, they weren't uniform and comprised a huge amount of not-quite-duplicate code. That's what the four dialog helper classes were intended to do. I take your point that they're not flexible enough.

ErrorDialog was aimed at providing ways of displaying both a bug dialog and dialogs where you could get the relevant stack trace for serious non-bug errors. It might have been a mistake to route WriteErrorDialog and ReadErrorDialog through ErrorDialog.I tend to see I/O errors as Errors, while the view among users seems to be that they're failures of a lesser sort.

I don't have a sense of which and how frequently the various methods in these classes are even used. Maybe I provided a lot which aren't at present, and it would be clearer if they were deprecated.